### PR TITLE
Chore/finish api

### PR DESCRIPTION
**Pull request by ChristophP**
A component can now be registerd like this:
```js
    WRC.register(MyComponent, 'my-component', [
      // these will be json parsed
      'name',
      'age',
      'fruits',
      'config',
      // specify a boolean attribute like this
      '!!isActive',
      // these will be handlers
      'onClickButton()',
      'onHoverName()',
    ]);

```

### CODE REVIEW

File: src/util.js [hunk 1]
```diff
@@ -14,6 +15,32 @@ export const mapObject = (fn, obj) => Object.keys(obj).reduce(
...
+// check if a propName corresponds to a boolean convention,
+// starting with "!!"
+export const isBoolConvention = prop => boolRegex.test(prop);
+
+// check if a propName corresponds to a handler convention,
+// ending in "()"
+export const isHandlerConvention = prop => handlerRegex.test(prop);
+
+// check if a propName corresponds to a handler convention,
+// ending in "()"
```
Comment (layflags) [inline src/util.js]: pls fix comment ;)

File: src/util.js [hunk 2]
```diff
@@ -14,6 +15,32 @@ export const mapObject = (fn, obj) => Object.keys(obj).reduce(
...
+, {});
+
+
+// ATTRIBUTE CONVENTIONS
+const boolRegex = /^!!/;
+const handlerRegex = /\(\)$/;
+
+// check if a propName corresponds to a boolean convention,
+// starting with "!!"
+export const isBoolConvention = prop => boolRegex.test(prop);
```
Comment (layflags) [inline src/util.js]: or `export const isBoolConvention = boolRegex.test.bind(boolRegex);`

Comment (ChristophP) [inline src/util.js]: bind isn't very functional :-/

File: src/util.js (hunk 1 above)
Comment (ChristophP) [inline src/util.js]: Done

**Comment by layflags** [comment]
Let's merge!

**Pull request merged**